service-workers/service-worker/active.https.html

```diff
@@ -5,51 +5,55 @@
+    await wait_for_state(t, registration.installing, 'activating');
+    const container = await frame.contentWindow.navigator.serviceWorker;
+    assert_equals(
+        container.controller,
+        null,
+        'On activating state a document should not have a controller');
+    assert_equals(
+        registration.active.scriptURL,
+        normalizeURL(SCRIPT),
+        'On activating state a document should have an active worker ');
+    assert_equals(
+        registration.waiting,
+        null,
+        'On activating state a document should not have a waiting worker');
+    assert_equals(
+        registration.installing,
+        null,
+        'On activating state a document should not have an installing worker');
 
-    service_worker_unregister(t, scope)
-      .then(step(function() { return with_iframe(scope); }))
-      .then(step(function(f) {
-          frame = f;
-          return navigator.serviceWorker.register(url, {scope: scope});
-        }))
-      .then(step(function(r) {
-          registration = r;
-          return wait_for_state(t, r.installing, 'activating');
-        }))
-      .then(step(function() {
-          var container = frame.contentWindow.navigator.serviceWorker;
-          assert_equals(
-            container.controller,
-            null,
-            'On activating state a document should not have a controller');
-          assert_equals(
-            registration.active.scriptURL,
-            normalizeURL(url),
-            'On activating state a document should have an active worker ');
-          assert_equals(
-            registration.waiting,
-            null,
-            'On activating state a document should not have a waiting worker');
-          assert_equals(
-            registration.installing,
-            null,
-            'On activating state a document should not have an installing ' +
-                'worker');
+    // FIXME: Add a test for a frame created after installation.
+    // Should the existing frame ("frame") block activation?
+    t.add_cleanup(() => frame.remove());
+    await registration.unregister();
+}, 'active is set');
 
-          // FIXME: Add a test for a frame created after installation.
-          // Should the existing frame ("frame") block activation?
-        }))
-      .then(step(function() {
-          frame.remove();
-          return service_worker_unregister_and_done(t, scope);
-        }))
-      .catch(unreached_rejection(t));
-  }, 'active is set');
+// Tests that the ServiceWorker objects returned from active attribute getter
+// that represent the same service worker are the same objects.
+promise_test(async t => {
+    registration =
+        await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+    await registration.active;
+    const r = await navigator.serviceWorker.getRegistration(SCOPE);
+    assert_equals(r.scriptURl, registration.scriptURl, 'getRegistration ' +
+                  'should return the same object');
```

Comment:
This isn't testing what the comments claim. It should be assert_equals(registration.active, r.active).

Ditto below.